src/chromo/common.py

```diff
@@ -573,6 +574,8 @@ def __init__(self, seed):
         self._rng = np.random.default_rng(seed)
         if hasattr(self._lib, "npy"):
             self._lib.npy.bitgen = self._rng.bit_generator.ctypes.bit_generator.value
+        self._apply_decay_handler = False
+        self._final_state_particles = np.array([], dtype=np.int64)
```

Comment:
This is array should not be there. You are duplicating state that exists in the model. You need to query the model implementation with model specific calles which particles it considers stable. Duplicating state is bad, because it tends to get out of sync.

Reply:
`self._final_state_particles` is set know through `final_state_particles` property and `set_stable` method.